src/utils/UserContext.jsx

```diff
@@ -0,0 +1,75 @@
+import { createContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { apiProvider } from '../api';
+import { useAuth } from '../hooks';
+
+export const UserContext = createContext({
+    /**
+     * @type {import('../api/types').User}
+     */
+    userDetails: null,
+
+    /**
+     * @type {import('../api/types').PostFull[]}
+     */
+    userPosts: null,
+
+    /**
+     * @type {import('../api/types').ChatRoomFull[]}
+     */
+    userChatRooms: null
+});
+
+export const UserContextProvider = ({ children }) => {
+    /**
+     * @type {import('../api/types').User}
+     */
+    const initialUserDetails = null;
+
+    /**
+     * @type {import('../api/types').PostFull[]}
+     */
+    const initialUserPosts = null;
+
+    /**
+     * @type {import('../api/types').ChatRoomFull[]}
```

Comment:
I like that you are putting some comments in here, but I wish that's the last PR that you are doing that :P 
I mean it's cool when no one is understing the code, but soon we will get to the point (at least we should) when the code should speak for itself, and the comments in code should be just for cases that are really hard to understand, so yeah, note for future, now it's fine I guess

Reply:
What's wrong with using JSDoc to specify types?

Reply:
In Real life project if its not an open source, the amount of comments in code its just a trash, when you get used to reading and understanding code, you dont need them, and for sure we wont be doing that in typescript :D